psl/builtin-connectors/src/postgres_datamodel_connector.rs

```diff
@@ -65,7 +65,8 @@ const CAPABILITIES: ConnectorCapabilities = enumflags2::make_bitflags!(Connector
     NativeUpsert |
     InsertReturning |
     UpdateReturning |
-    RowIn
+    RowIn |
+    Distinct
```

Comment:
Two points:

1. Doesn't CockroachDB also support `DISTINCT ON`?
2. I'd rename that to `DistinctOn` specifically. Because most connectors support `DISTINCT` but that's not what we're using here.

Reply:
This has been renamed to `DistinctOn` :)

I've now tested CockroachDB and it works ✨ however it returns differently ordered result sets (same actual data) for the following tests (as they do not use `orderBy`) breaking their snapshots
<img width="349" alt="image" src="https://github.com/prisma/prisma-engines/assets/29753584/171af49e-a99c-442b-aec6-ad0492a4e1d7">

Reply:
This has been fixed per the latest commit with the addition of extra tests to track current differing snapshots using ` #[connector_test(exclude(...))]`